### PR TITLE
Enhance Money with service charge, GST calculation

### LINKED ISSUE #108: Adding + or ++ to money field automatically adds GST and service tax
As a user, I can specify if an amount of money in the debt is inclusive or exclusive of GST and service charge, so that I do not have to manually calculate the final debt amount.

### PR DESCRIPTION
**Pull request by waynezsy**
Resolves #108

### CODE REVIEW

**Review by lyuting47 — APPROVED** [review]
LGTM

File: src/main/java/paymelah/model/debt/Money.java [hunk 1]
```diff
@@ -29,7 +33,24 @@ public Money(String money) {
...
+        }
+
+        BigDecimal value = new BigDecimal(money).setScale(CENTS_PRECISION);
+        if (plusCount == 1) {
+            value = value.multiply(GST_MULTIPLIER).setScale(CENTS_PRECISION, RoundingMode.CEILING);
+        } else if (plusCount == 2) {
+            value = value.multiply(SERVICE_CHARGE_MULTIPLIER).multiply(GST_MULTIPLIER)
+                    .setScale(CENTS_PRECISION, RoundingMode.CEILING);
+        } else if (plusCount != 0) {
+            assert false : ">2 '+' not detected as invalid Money input.";
```
Comment (tsammeow) [inline src/main/java/paymelah/model/debt/Money.java:50]: Should this be an assert? This is not a developer error, maybe we should throw a ParseException instead

**Review by tsammeow — CHANGES_REQUESTED** [review]

File: src/main/java/paymelah/model/debt/Money.java (hunk 1 above)
Comment (tsammeow) [reply src/main/java/paymelah/model/debt/Money.java:50]: Never mind, it should not be able to reach here with more than 2 +s

**Review by tsammeow — APPROVED** [review]
LGTM!

File: src/main/java/paymelah/model/debt/Money.java (hunk 1 above)
Comment (waynezsy) [reply src/main/java/paymelah/model/debt/Money.java:50]: The full code (hidden by the diff) already checks if the `money:String` is valid and throws an exception if it is not, so if it enters this if block it should be a dev error.

**Pull request merged**